### ISSUE TITLE
Option D: Fine-tune on Q&A Benchmark (Small but Relevant Data)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#247]
## Problem
We have 1M+ triplets from document co-occurrence, but this training signal doesn't match retrieval. Meanwhile, we have 50 Q&A benchmark questions with gold answers that directly represent the task we care about.

## Proposed Solution
Fine-tune embeddings on the small but directly relevant Q&A benchmark data.

### Key Insight
**Quality > Quantity** for training data. 50 real query-document pairs are more valuable than 1M proxy triplets.

### Approach
1. **Extract training pairs from Q&A benchmark**:
   - Query: the question text
   - Positive: document(s) containing the gold answer
   - Negative: documents that don't contain the answer

2. **Data augmentation** to expand 50 → ~500-1000 examples:
   - Paraphrase questions (synonym substitution using linguistic embeddings)
   - Multiple positive docs per question (if answer appears in several)
   - Hard negatives: docs with similar roots but wrong answer

3. **Fine-tuning strategy**:
   - Start from pre-trained linguistic embeddings
   - Small learning rate (don't destroy general knowledge)
   - Early stopping on held-out questions

### Training
```python
for question, positive_doc, negative_doc in triplets:
    q_embed = embed_query(question)
    pos_embed = embed_doc(positive_doc)
    neg_embed = embed_doc(negative_doc)
    
    loss = triplet_loss(q_embed, pos_embed, neg_embed, margin=0.3)
```

### Advantages
- **Direct task alignment**: Training on exactly what we're evaluating
- **Interpretable**: Can trace which examples helped/hurt
- **Fast**: Small dataset = quick iteration
- **Grounded**: Real questions, not synthetic

### Risks
- **Overfitting**: 50 questions is small
- **Distribution shift**: Benchmark may not represent all queries
- **Leakage**: Must hold out test set properly

### Mitigation
- Cross-validation (train on 40, test on 10, rotate)
- Monitor train/test gap closely
- Combine with other approaches (C, then D)

### Implementation Steps
1. [ ] Parse Q&A benchmark to extract query-doc pairs
2. [ ] Find which documents contain each answer
3. [ ] Implement augmentation (paraphrase, hard negatives)
4. [ ] Build fine-tuning pipeline with cross-validation
5. [ ] Train and evaluate

## Compatibility
- **Best after**: Option C (establish baseline first)
- **Can combine with**: Option A (synthetic questions expand data)
- **Orthogonal to**: Option B (different architecture)

## Estimated Effort
Medium: 1-2 days

---
Related to #243 (topical embeddings discrimination issue)

**Comment by marctjones** [comment]
Benchmark too small for fine-tuning. Focus on corpus quality instead (#411).
